nomad/structs/acl.go

```diff
@@ -232,6 +233,27 @@ func (a *ACLToken) IsExpired(t time.Time) bool {
 	return a.ExpirationTime.Before(t) || t.IsZero()
 }
 
+// RoleSubset checks if a given set of role IDs are assigned to the ACL token.
+func (a *ACLToken) RoleSubset(roleIDs []string) bool {
+
+	// Hot-path: management tokens allows access to all roles.
+	if a.Type == ACLManagementToken {
+		return true
+	}
```

Comment:
It looks like this is safe for the consumers we have today. But a role can have a policy with `deny` as well. Maybe we shouldn't have this bit of convenience and _require_ that any consumers of this are explicitly checking the management role instead, so that future consumers can't accidentally lock out the management token from something?

Reply:
Yeah this makes perfect sense and is a better approach. I have made the change and updated the function comment to call this out within commit: https://github.com/hashicorp/nomad/pull/14923/commits/3481a620406ba5b1f2c4d1c29047e0a9d82cef67